src/core/strategies/token.js

```diff
@@ -0,0 +1,285 @@
+     * @returns {tokenChange|null} - The result of the strategy: `null` if cannot apply, or the descriptor of the change
+     */
+    function strategyIdentifier(index, tokens) {
+        let token = tokens[index] || null;
+        let type = tokensHelper.getType(token);
+        let result = null;
+
+        if (tokensHelper.isIdentifier(type) && index >= 0) {
+            if (index === 0) {
+                // the token is the first of the expression, so the sign is implicit +, simply negate the value
+                result = insertNegativeSign(token);
+            } else {
+                // the token is preceded by something else
+                result = applySignChange(index - 1, tokens);
+            }
+        }
+
+        return result;
+    },
+
+    /**
+     * Strategy that applies on sub-expression only
+     * @param {number} index - The index of the current token
+     * @param {token[]} tokens - The list of tokens that represent the expression
+     * @returns {tokenChange|null} - The result of the strategy: `null` if cannot apply, or the descriptor of the change
+     */
+    function strategyExpression(index, tokens) {
+        let token = tokens[index] || null;
+        let type = tokensHelper.getType(token);
+        let result = null;
+        let count = 0;
+
+        if (tokensHelper.isAggregator(type) && index >= 0) {
+            if (token.type === 'RPAR') {
+                count++;
+            }
+
+            // find the opening parenthesis
+            while (index && (token.type !== 'LPAR' || count)) {
+                index--;
+                token = tokens[index] || null;
+
+                if (token.type === 'RPAR') {
+                    count++;
+                }
+                if (token.type === 'LPAR') {
+                    count--;
+                }
+            }
+
+            if (!count && token.type === 'LPAR') {
+                if (index === 0) {
+                    // the token is the first of the expression, so the sign is implicit +, simply negate the value
+                    result = insertNegativeSign(token);
+                } else {
+                    // the token is preceded by something else
+                    result = applySignChange(index - 1, tokens);
+                }
+            }
+        }
+
+        return result;
+    }
+];
+
+/**
+ * Apply a list of strategies to a token.
+ * @param {number} index - The index of the current token
+ * @param {token[]} tokens - The list of tokens that represent the expression
+ * @param {tokenStrategy[]} strategies - The list of strategies to apply.
+ * @returns {tokenChange|null} - The result of the strategy: `null` if cannot apply, or the descriptor of the change
+ */
+export function applyTokenStrategies(index, tokens, strategies) {
+    let result = null;
+
+    strategies.every(strategy => {
+        result = strategy(index, tokens);
+        return !result;
```

Comment:
hmmm what about return !strategy(index, tokens) ? :)

Reply:
nope, because this is a side effect function :). The value of result is used in 2 places: here for stopping the iteration, and later for giving the result back to the caller of `applyTokenStrategies`